File: OPENAPI-REST-API/swagger-client/spring/src/main/java/io/swagger/model/VpsOrderPlatformPackages.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.v3.oas.annotations.media.Schema;
import java.math.BigDecimal;
import org.springframework.validation.annotation.Validated;
import org.openapitools.jackson.nullable.JsonNullable;
import io.swagger.configuration.NotUndefined;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Platform Packages
 */
@Schema(description = "Platform Packages")
@Validated
@NotUndefined



public class VpsOrderPlatformPackages   {
  @JsonProperty("kvm")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private BigDecimal kvm = null;

  @JsonProperty("kvmstorage")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private BigDecimal kvmstorage = null;

  @JsonProperty("hyperv")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private BigDecimal hyperv = null;


  public VpsOrderPlatformPackages kvm(BigDecimal kvm) { 

    this.kvm = kvm;
    return this;
  }

  /**
   * Get kvm
   * @return kvm
   **/
  
  @Schema(example = "32", description = "")
  
@Valid
  public BigDecimal getKvm() {  
    return kvm;
  }



  public void setKvm(BigDecimal kvm) { 
    this.kvm = kvm;
  }

  public VpsOrderPlatformPackages kvmstorage(BigDecimal kvmstorage) { 

    this.kvmstorage = kvmstorage;
    return this;
  }

  /**
   * Get kvmstorage
   * @return kvmstorage
   **/
  
  @Schema(example = "57", description = "")
  
@Valid
  public BigDecimal getKvmstorage() {  
    return kvmstorage;
  }



  public void setKvmstorage(BigDecimal kvmstorage) { 
    this.kvmstorage = kvmstorage;
  }

  public VpsOrderPlatformPackages hyperv(BigDecimal hyperv) { 

    this.hyperv = hyperv;
    return this;
  }

  /**
   * Get hyperv
   * @return hyperv
   **/
  
  @Schema(example = "54", description = "")
  
@Valid
  public BigDecimal getHyperv() {  
    return hyperv;
  }



  public void setHyperv(BigDecimal hyperv) { 
    this.hyperv = hyperv;
  }

  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    VpsOrderPlatformPackages vpsOrderPlatformPackages = (VpsOrderPlatformPackages) o;
    return Objects.equals(this.kvm, vpsOrderPlatformPackages.kvm) &&
        Objects.equals(this.kvmstorage, vpsOrderPlatformPackages.kvmstorage) &&
        Objects.equals(this.hyperv, vpsOrderPlatformPackages.hyperv);
  }

  @Override
  public int hashCode() {
    return Objects.hash(kvm, kvmstorage, hyperv);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class VpsOrderPlatformPackages {\n");
    
    sb.append("    kvm: ").append(toIndentedString(kvm)).append("\n");
    sb.append("    kvmstorage: ").append(toIndentedString(kvmstorage)).append("\n");
    sb.append("    hyperv: ").append(toIndentedString(hyperv)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: OPENAPI-REST-API/openapi-client/cpp-tiny/lib/Models/VpsOrderPlatformPackages.cpp


#include "VpsOrder_platformPackages.h"

using namespace Tiny;

VpsOrder_platformPackages::VpsOrder_platformPackages()
{
	kvm = float(0);
	kvmstorage = float(0);
	hyperv = float(0);
}

VpsOrder_platformPackages::VpsOrder_platformPackages(std::string jsonString)
{
	this->fromJson(jsonString);
}

VpsOrder_platformPackages::~VpsOrder_platformPackages()
{

}

void
VpsOrder_platformPackages::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *kvmKey = "kvm";

    if(object.has_key(kvmKey))
    {
        bourne::json value = object[kvmKey];



        jsonToValue(&kvm, value, "long");


    }

    const char *kvmstorageKey = "kvmstorage";

    if(object.has_key(kvmstorageKey))
    {
        bourne::json value = object[kvmstorageKey];



        jsonToValue(&kvmstorage, value, "long");


    }

    const char *hypervKey = "hyperv";

    if(object.has_key(hypervKey))
    {
        bourne::json value = object[hypervKey];



        jsonToValue(&hyperv, value, "long");


    }


}

bourne::json
VpsOrder_platformPackages::toJson()
{
    bourne::json object = bourne::json::object();





    object["kvm"] = getKvm();






    object["kvmstorage"] = getKvmstorage();






    object["hyperv"] = getHyperv();



    return object;

}

long
VpsOrder_platformPackages::getKvm()
{
	return kvm;
}

void
VpsOrder_platformPackages::setKvm(long kvm)
{
	this->kvm = kvm;
}

long
VpsOrder_platformPackages::getKvmstorage()
{
	return kvmstorage;
}

void
VpsOrder_platformPackages::setKvmstorage(long kvmstorage)
{
	this->kvmstorage = kvmstorage;
}

long
VpsOrder_platformPackages::getHyperv()
{
	return hyperv;
}

void
VpsOrder_platformPackages::setHyperv(long hyperv)
{
	this->hyperv = hyperv;
}



